src/cuda/common.cu

```diff
@@ -48,19 +48,18 @@ __global__ void fseries_kernel_compute(int nf1, int nf2, int nf3, T *f,
 
   for (int i = blockDim.x * blockIdx.x + threadIdx.x; i < nf / 2 + 1;
        i += blockDim.x * gridDim.x) {
-    int brk = 0.5 + i;
-    T x     = 0.0;
+    T x = 0.0;
     for (int n = 0; n < q; n++) {
-      x += ft[n] * 2 * (pow(cabs(at[n]), brk) * cos(brk * carg(at[n])));
+      x += ft[n] * 2 * cos(i * at[n]);
```

Comment:
Together with @blackwer, finding:
@ahbarnett the magnitude of a pure imaginary number is always 1 -> pow of that is also always one so we can get rid of it entirely. The carg is using atan2 to compute the polar angle (undoing the exp from before) hence we can not do the exp in the pre-compute step and not do the atan2 here. 

This gives back the analytical formula?  Now for type 3 since the grid is not regular i (formerly brk) needs to be replaced with `k[j]`